internal/provider/kubernetes/gatewayclass.go

```diff
@@ -96,21 +102,27 @@ func (r *gatewayClassReconciler) Reconcile(ctx context.Context, request reconcil
 
 	var cc controlledClasses
 
+	found := false
 	for i := range gatewayClasses.Items {
 		if gatewayClasses.Items[i].Spec.ControllerName == r.controller {
 			cc.addMatch(&gatewayClasses.Items[i])
+			if gatewayClasses.Items[i].GetName() == request.Name {
```

Comment:
can we simplify the logic here
```
if gatewayClasses.Items[i].Spec.ControllerName == r.controller {
    cc.addMatch(&gatewayClasses.Items[I])
    r.resourceTable.GatewayClasses.Store(gc.GetName(), gc)
} else {
    r.resourceTable.GatewayClasses.Delete(request.Name)
}    
```

Reply:
`i` may not be the same gatewayclass as the `request`.

```if gatewayClasses.Items[i].GetName() == request.Name {```
^ ensures that only the gatewayclass that matches the request is added to the resource table.

Reply:
does that mater here ? imho below logic determines if we are interested in the gateway class or not
```
if gatewayClasses.Items[i].Spec.ControllerName == r.controller {
```

Reply:
`i` may be different than the request if multiple gc's exist with EG's configured controller name.